zerver/actions/create_user.py

```diff
@@ -320,13 +322,25 @@ def notify_created_user(user_profile: UserProfile) -> None:
         "custom_profile_field_data": {},
     }
 
-    active_users = user_profile.realm.get_active_users()
+    user_ids_without_access_to_created_user: List[int] = []
+    users_with_access_to_created_users: List[UserProfile] = []
+    active_realm_users = list(user_profile.realm.get_active_users())
+
+    if user_access_restricted_in_realm(user_profile):
+        for user in active_realm_users:
+            if user.is_guest:
+                user_ids_without_access_to_created_user.append(user.id)
+            else:
+                users_with_access_to_created_users.append(user)
```

Comment:
This block should have a comment:
# This logic assumes that `can_access_all_users_group` is EVERYONE or MEMBERS.

Reply:
Added the comment.